CDTDatastore/query/CDTQIndex.h

```diff
@@ -56,7 +65,12 @@ extern NSString *const kCDTQTextType;
 + (nullable instancetype)index:(NSString *)indexName
                     withFields:(NSArray<NSString *> *)fieldNames
                         ofType:(NSString *)indexType
-                  withSettings:(nullable NSDictionary<NSString *, NSString *> *)indexSettings;
+                  withSettings:(nullable NSDictionary<NSString *, NSString *> *)indexSettings __deprecated;
+
++ (nullable instancetype)index:(NSString *)indexName
```

Comment:
This method should probably have its own doc comments (largely copied from the deprecated method).